actions/views/channel.js

```diff
@@ -100,20 +102,32 @@ export function leaveChannel(channelId) {
         const state = getState();
         const myPreferences = getMyPreferences(state);
         const currentUserId = getCurrentUserId(state);
-        const currentTeamId = getCurrentTeamId(state);
+        const currentTeam = getCurrentTeam(state);
 
         if (isFavoriteChannel(myPreferences, channelId)) {
             dispatch(unfavoriteChannel(channelId));
         }
 
         const teamUrl = getCurrentRelativeTeamUrl(state);
-        LocalStorageStore.removePreviousChannelName(currentUserId, currentTeamId);
-        browserHistory.push(teamUrl);
-
+        LocalStorageStore.removePreviousChannelName(currentUserId, currentTeam.id);
         const {error} = await dispatch(leaveChannelRedux(channelId));
         if (error) {
             return {error};
         }
+        const prevChannelName = LocalStorageStore.getPreviousChannelName(currentUserId, currentTeam.id);
+        const channelsInTeam = getChannelsNameMapInCurrentTeam(state);
+        const prevChannel = getChannelByName(channelsInTeam, prevChannelName);
+        if (!prevChannel || !getMyChannelMemberships(getState())[prevChannel.id]) {
+            LocalStorageStore.removePreviousChannelName(currentUserId, currentTeam.id);
```

Comment:
Any reason why `removePreviousChannelName` is called again? Is it because of the dispatch to `leaveChannelRedux` ?

Reply:
Yes, the reason is because you can end up having a channel there that is not a channel that you are a member of, so... if that the case, I'll ignore it to avoid the error window saying you that you can't access the channel.

Reply:
To give you more context, you have the last viewed channel and the penultimate one, so, I only have to check it twice to be sure.

Reply:
👍